og.module

```diff
@@ -377,6 +377,25 @@ function _og_form_group_reference_validate($form, &$form_state) {
   form_set_error('og', t('You must select one or more groups for this content.'));
 }
 
+
+/**
+ * Implements hook_field_formatter_info_alter()
+ *
+ * Allow OG audience fields to have entity reference formatters.
+ */
+function og_field_formatter_info_alter(array &$info) {
+  foreach (array_keys($info) as $key) {
+    if (strpos($key, 'entity_reference_') !== 0) {
+      // Not an entity reference formatter.
+      continue;
+    }
```

Comment:
A better check to see if this formatter targets entity reference fields is to inspect the `field_types`:

```
if (!in_array('entity_reference', $info['field_types'])) {
  continue;
}
```